core/dnsserver/onstartup.go

```diff
@@ -2,15 +2,27 @@ package dnsserver
 
 import (
 	"fmt"
+	"regexp"
 	"sort"
 )
 
+// Below checkDomainName() function checks whether the given domain-name is following RFC1035 preferred syntax or not.
+// If not correct then shows a warning against domain-name.
+func checkDomainName(addr string) bool {
+	matchDomain, _ := regexp.Compile("[A-Za-z]+://([A-Za-z0-9-]+\\.[A-Za-z0-9-]+)*:[0-9]+")
```

Comment:
This regex should only match a domain name, not the protocol prefix and port suffix.